contentlayer.config.ts

```diff
@@ -1,18 +1,112 @@
-import { defineDocumentType, makeSource, ComputedFields } from "contentlayer2/source-files";
-
-// this is a basic example to show how to structure your content source
-export const _DataName = defineDocumentType(() => ({
-  name: "Post", // name of the post
-  filePathPattern: `**/*.md`, // file path where the mdx file to be process is located
-  contentType: "mdx", // content type mdx
-  // fields: types desc
+import * as fs from "fs";
+import { defineDocumentType, defineNestedType, makeSource } from "contentlayer2/source-files";
+
+const Resources = defineNestedType(() => ({
+  name: "Resources",
   fields: {
-    title: { type: "string", required: true },
-    date: { type: "date", required: true },
-  },
-  computedFields: {
-    url: { type: "string", resolve: (post) => `/posts/${post._raw.flattenedPath}` },
+    title: { type: "string" },
+    url: { type: "string" },
   },
 }));
 
-export default makeSource({ contentDirPath: "posts", documentTypes: [_DataName] });
+export const generateDocuTypes = () => {
+  const getFolders = fs.readdirSync(`${process.cwd()}/public/bitcoin-transcript`, "utf-8");
+  const folders = getFolders.slice(6).filter((item) => item !== "twitter_handles.json");
+
+  const DefinedDocumentTypes = folders.map((name) => {
+    const slugifyName = name
+      .split("-")
+      .map((text) => text.replace(/(^\w{1})|(\s+\w{1})/g, (text) => text.toUpperCase()))
```

Comment:
let give this regex a descriptive variable name (perhaps with an additional comment) so its clear what the regex is doing